### PR TITLE
Add definition for trtc-js-sdk

### PR DESCRIPTION
**Pull request by yokots**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/trtc-js-sdk/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,756 @@
+// Type definitions for trtc-js-sdk 4.3
+// Project: https://github.com/tencentyun/TRTCSDK#readme
+// Definitions by: yokots <https://github.com/yokots>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/*~ If this module is a UMD module that exposes a global variable 'myLib' when
```
Comment (armanio123) [inline types/trtc-js-sdk/index.d.ts:6]: Remove this comment.

File: types/trtc-js-sdk/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,756 @@
...
+ *~ loaded outside a module loader environment, declare that global here.
+ *~ Otherwise, delete this declaration.
+ */
+// tslint:disable-next-line:export-just-namespace
```
Comment (armanio123) [inline types/trtc-js-sdk/index.d.ts:10]: Could you please enable the tslint rule back. There's no need to export only a namespace. Instead use export on the objects you want to export.

**Review by armanio123 — CHANGES_REQUESTED** [review]

File: types/trtc-js-sdk/index.d.ts (hunk 2 above)
Comment (yokots) [reply types/trtc-js-sdk/index.d.ts:6]: this is an umd module

Comment (yokots) [reply types/trtc-js-sdk/index.d.ts:6]: I also enable esModuleInterop in tsconfig.json

Comment (armanio123) [reply types/trtc-js-sdk/index.d.ts:6]: The `export as namespace` already does the job of exporting for umd. Grouping all of the interfaces inside the a `namespace` is unnecessary. You can read more about the rule on the tslint site.

Comment (yokots) [reply types/trtc-js-sdk/index.d.ts:6]: when remove `export = xxx`,  the typescript report some errors, `Global module exports may only appear in module files.` and `Duplicate identifier Omit`, I write this definition just like `react` and other modules, I don't known how to fix it. And apologize for my poor english.

Comment (armanio123) [reply types/trtc-js-sdk/index.d.ts:6]: Sorry for not being descriptive enough. You'll need to remove `export =` and `declare namespace TRTC`, as you are already exporting the namespace by using `export as namespace TRTC`. Then, prepend `export` on every object the library has exports like `export namespace Logger`, `export function checkSystemRequirements(): Promise<boolean>;`, etc.

Hope this makes sense.

Comment (yokots) [reply types/trtc-js-sdk/index.d.ts:6]: Thanks for Detailed explanation，I have fixed it.

**Review by armanio123 — APPROVED** [review]

**Pull request merged**